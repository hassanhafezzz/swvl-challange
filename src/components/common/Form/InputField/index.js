import React from 'react';
import { useField } from 'formik';

import classNames from 'classnames/bind';
import styles from './styles.module.css';

const cx = classNames.bind(styles);

const InputField = ({ id, name, type, label }) => {
  const [field, meta] = useField(name);
  return (
    <>
      <label
        htmlFor={id}
        className={cx('label', meta.error && meta.touched ? 'label-error' : '')}
      >
        {label}
      </label>

      <input
        {...field}
        id={id}
        name={name}
        type={type}
        className={cx(
          'input',
          meta.error && meta.touched ? 'error-field' : null,
        )}
      />

      {meta.error && meta.touched ? (
        <p className={cx('error')}>{meta.error}</p>
      ) : null}
    </>
  );
};

export default InputField;
