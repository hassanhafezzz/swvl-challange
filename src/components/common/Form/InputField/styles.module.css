.label {
  margin: 20px 0 8px;
  display: flex;
  text-transform: capitalize;
  font-size: var(--small-paragraph);
  font-weight: var(--bold);
}

.label-error {
  color: var(--color-red);
}

.input {
  position: relative;
  display: block;
  margin: 0;
  width: 100%;
  height: 45px;
  padding: 10px 15px;
  font-size: var(--paragraph-font);
  color: var(--color-black);
  background-color: var(--color-white);
  border: 2px solid var(--color-gray-300);
  border-radius: 4px;
}

.error-field {
  border: 2px solid var(--color-red);
}

.input:hover,
.input:focus {
  border: 2px solid var(--color-light-secondary);
  outline: none;
}

.error {
  margin: 5px 0 0;
  display: flex;
  color: var(--color-red);
  font-size: var(--small-paragraph-font);
}
