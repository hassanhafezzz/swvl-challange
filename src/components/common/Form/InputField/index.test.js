import React from 'react';
import { Formik } from 'formik';
import { render } from '@testing-library/react';

import InputField from './index';

const component = (
  <Formik>
    <InputField type="text" name="name" id="name" label="name" />
  </Formik>
);

test('input should have label', () => {
  const { getByLabelText } = render(component);
  const input = getByLabelText('name');
  expect(input).toBeTruthy();
});

test('input snapshot', () => {
  const { container, asFragment } = render(component);
  expect(asFragment(container)).toMatchSnapshot();
});
