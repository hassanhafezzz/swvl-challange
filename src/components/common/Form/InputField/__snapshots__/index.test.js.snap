// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`input snapshot 1`] = `
<DocumentFragment>
  <label
    class="label"
    for="name"
  >
    name
  </label>
  <input
    class="input"
    id="name"
    name="name"
    type="text"
    value=""
  />
</DocumentFragment>
`;
